### ISSUE TITLE
MBLayout error in LSTMP training

### ISSUE THREAD

**Issue by Turan111** [Microsoft/CNTK#412]
Hi,

I get following error when training LSTMP on 20hr corpus at 5th epoch:

`EXCEPTION occurred: MBLayout: Attempting to read out flags, but only only 255 out of 256 frames have been defined.
`
My config is the same as AN4 LSTMP config. Full error message is as following:

```
Epoch[ 5 of 40]-Minibatch[ 741- 750, 0.000000000000260%]: SamplesSeen = 2560; TrainLossPerSample =  1.18583984; EvalErr[0]PerSample = 0.34101562; TotalTime = 1.3488s; SamplesPerSecond = 1897.9


[CALL STACK]
  /usr/bin/../lib/libcntkmath.so ( Microsoft::MSR::CNTK::DebugUtil::PrintCallStack()  + 0xbf  )  [0x7f3ef98b8b1f]
  cntk                           ( void Microsoft::MSR::CNTK::ThrowFormatted<std::logic_error>(char const*, ...)  + 0xdd  )  [0x53d99d]
  cntk                           ( Microsoft::MSR::CNTK::MBLayout::IsBeyondStartOrEnd(Microsoft::MSR::CNTK::FrameRange const&) const  + 0xfa  )  [0x5ca9aa]
  cntk                           ( Microsoft::MSR::CNTK::DelayedValueNodeBase<float, -1>::ForwardProp(Microsoft::MSR::CNTK::FrameRange const&)  + 0xc1  )  [0x600d81]
  cntk                           ( Microsoft::MSR::CNTK::ComputationNetwork::SEQTraversalFlowControlNode::ForwardProp(Microsoft::MSR::CNTK::FrameRange const&)  + 0xbc  )  [0x6b562c]
  cntk                           ( Microsoft::MSR::CNTK::ComputationNetwork::PARTraversalFlowControlNode::ForwardProp(Microsoft::MSR::CNTK::FrameRange const&)  + 0xd3  )  [0x6b54b3]
  cntk                           ( Microsoft::MSR::CNTK::ComputationNetwork::ForwardProp(std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>)  + 0x7c  )  [0x6b787c]
  cntk                           ( Microsoft::MSR::CNTK::SGD<float>::TrainOneEpoch(std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> const&, int, unsigned long, Microsoft::MSR::CNTK::IDataReader<float>*, double, unsigned long, std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::map<std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >, Microsoft::MSR::CNTK::Matrix<float>*, std::less<std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > >, std::allocator<std::pair<std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > const, Microsoft::MSR::CNTK::Matrix<float>*> > >*, std::list<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> > > const&, std::list<Microsoft::MSR::CNTK::Matrix<float>, std::allocator<Microsoft::MSR::CNTK::Matrix<float> > >&, double&, std::vector<double, std::allocator<double> >&, unsigned long&, std::string)  + 0xa0c )  [0x73842c]
  cntk                           ( Microsoft::MSR::CNTK::SGD<float>::TrainOrAdaptModel(int, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>, bool, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>, std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>, Microsoft::MSR::CNTK::IDataReader<float>*, Microsoft::MSR::CNTK::IDataReader<float>*)  + 0x1244)  [0x73d2f4]
  cntk                           ( Microsoft::MSR::CNTK::SGD<float>::Train(std::function<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork> (int)>, int, Microsoft::MSR::CNTK::IDataReader<float>*, Microsoft::MSR::CNTK::IDataReader<float>*, bool)  + 0x424 )  [0x73f0f4]
  cntk                           ( void DoTrain<Microsoft::MSR::CNTK::ConfigParameters, float>(Microsoft::MSR::CNTK::ConfigParameters const&)  + 0x22c )  [0x75727c]
  cntk                           ( void DoCommands<float>(Microsoft::MSR::CNTK::ConfigParameters const&)  + 0x81c )  [0x59446c]
  cntk                           ( wmainOldCNTKConfig(int, wchar_t**)        + 0x909 )  [0x535ab9]
  cntk                           ( wmain1(int, wchar_t**)                    + 0x68  )  [0x536108]
  cntk                           ( main                                      + 0xd8  )  [0x529aa8]
  /lib/x86_64-linux-gnu/libc.so.6 ( __libc_start_main                         + 0xf5  )  [0x7f3ef8524ec5]
  cntk                           (                                                   )  [0x52da47]
EXCEPTION occurred: MBLayout: Attempting to read out flags, but only only 255 out of 256 frames have been defined.
```

**Comment by yzhang87** [comment]
Which reader are you using?

**Comment by Turan111** [comment]
My config is the same as AN4 LSTMP config. 
https://github.com/Microsoft/CNTK/blob/master/Examples/Speech/AN4/Config/LSTM-NDL.cntk

```
reader = [
        readerType = "HTKMLFReader"
        readMethod = "blockRandomize"
        miniBatchMode = "partial"
        randomize = "auto"
```

**Comment by Turan111** [comment]
When I set as
```
minibatchSize = 10
nbrUttsIneachRecurrentIter = 10
```
or
```
minibatchSize = 10
nbrUttsIneachRecurrentIter = 16

```
training is going normal.  Problem arises when I take 
```
minibatchSize = 16
nbrUttsIneachRecurrentIter = 16
```

**Comment by NewEricWang** [comment]
Hi, Turan11
      I also ran into similar problem when training LSTMP on 100hr corpus. My NDL of LSTM is the same as AN4 LSTMP " "lstmp-3layer-opt.ndl.  I set as 
minibatchSize = 256
nbrUttsIneachRecurrentIter = 256

I get following error before starting 1th epoch:
EXCEPTION occurred: MBLayout: Attempting to read out flags, but only only 65241 out of 65536 frames have been defined.

[CALL STACK]
[0x5fe895]     
[0x660139]          Microsoft::MSR::CNTK::MeanNode<float>::  ForwardPropNonLooping  () + 0xd9 
[0x667943]          Microsoft::MSR::CNTK::ComputationNetwork::PARTraversalFlowControlNode::  ForwardProp  (Microsoft::MSR::CNTK::FrameRange const&) + 0xd3 
[0x669e7c]          Microsoft::MSR::CNTK::ComputationNetwork::  ForwardProp  (std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>) + 0x7c 
[0x6f8d54]          Microsoft::MSR::CNTK::SGD<float>::  PreCompute  (std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>,  Microsoft::MSR::CNTK::IDataReader*,  std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>,std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>>>&,  std::vector<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>,std::allocator<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>>>&,  Microsoft::MSR::CNTK::StreamMinibatchInputs*) + 0x2f4
[0x6fda9d]          Microsoft::MSR::CNTK::SGD<float>::  TrainOrAdaptModel  (int,  std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>,  bool,  std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork>,  std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase>,  Microsoft::MSR::CNTK::IDataReader*,  Microsoft::MSR::CNTK::IDataReader*) + 0x99d
[0x70030d]          Microsoft::MSR::CNTK::SGD<float>::  Train  (std::function<std::shared_ptr<Microsoft::MSR::CNTK::ComputationNetwork> (int)>,  int,  Microsoft::MSR::CNTK::IDataReader*,  Microsoft::MSR::CNTK::IDataReader*,  bool) + 0x44d
[0x705fa2]          void  DoTrain  <Microsoft::MSR::CNTK::ConfigParameters,float>(Microsoft::MSR::CNTK::ConfigParameters const&) + 0x362
[0x5ad5cb]          void  DoCommands  <float>(Microsoft::MSR::CNTK::ConfigParameters const&) + 0x8ab
[0x54fed9]          wmainOldCNTKConfig  (int,  wchar_t**)              + 0x899
[0x5504f8]          wmain1  (int,  wchar_t**)                          + 0x68 
[0x544ac8]          main                                               + 0xd8 
[0x7fd14da35b15]    __libc_start_main                                  + 0xf5 
[0x548775]

**Comment by wolfma61** [comment]
The issue has been dormant for a long while. Please open a new issue if you still experience problems in the latest release.
